lib/src/widgets/choice_button.dart

```diff
@@ -1,41 +1,73 @@
+import 'dart:ui';
+
 import 'package:cap_countdown/src/exam/optional_question.dart';
 import 'package:cap_countdown/src/exam/question_choice.dart';
+import 'package:cap_countdown/src/util/events_enum.dart';
 import 'package:flutter/material.dart';
 import 'package:latext/latext.dart';
 
 class ChoiceButton extends StatelessWidget {
   final QuestionChoice choice;
   final OptionalQuestion question;
   final bool submitted;
+  final bool isCrossOut;
   final ValueChanged<QuestionChoice?>? onChanged;
+  final ValueChanged<EventsEnum?>? onEvent;
 
   const ChoiceButton(
       {super.key,
       required this.choice,
       required this.question,
       required this.submitted,
-      this.onChanged});
+      required this.isCrossOut,
+      this.onChanged,
+      this.onEvent});
 
   @override
   Widget build(BuildContext context) {
-    return RadioListTile<QuestionChoice>(
-      // Use LaTexT to render LaTeX (math formula) in text.
-      title: LaTexT(
-          laTeXCode:
-              Text('(${choice.answer.name}) ${choice.description ?? ''}')),
-      value: choice,
-      groupValue: question.selectedChoice,
-      fillColor: _getFillColor(),
-      onChanged: (value) {
-        // If the question has been submitted, can't change the answer.
-        if (submitted) return;
-
-        if (value == question.selectedChoice) {
-          onChanged?.call(null);
-        } else {
-          onChanged?.call(value);
+    final text = '(${choice.answer.name}) ${choice.description ?? ''}';
+
+    return GestureDetector(
+      onHorizontalDragEnd: (DragEndDetails dragEndDetails) {
+        if (dragEndDetails.velocity.pixelsPerSecond.dx < 0) {
+          onEvent?.call(EventsEnum.crossOutChoice);
         }
       },
+      supportedDevices: const <PointerDeviceKind>{PointerDeviceKind.touch},
+      behavior: HitTestBehavior.opaque,
+      excludeFromSemantics: true,
+      child: RadioListTile<QuestionChoice>(
+        // Use LaTexT to render LaTeX (math formula) in text.
+        title: LaTexT(
+            laTeXCode: Text(
+          text,
+          style: isCrossOut
+              ? const TextStyle(
+                  decoration: TextDecoration.lineThrough, color: Colors.grey)
+              : null,
+        )),
+        value: choice,
+        groupValue: question.selectedChoice,
+        fillColor: _getFillColor(),
+        onChanged: (value) {
+          // If the question has been submitted, can't change the answer.
+          if (submitted) return;
+          if (isCrossOut) {
+            ScaffoldMessengerState messenger = ScaffoldMessenger.of(context);
+            messenger.clearSnackBars();
+            messenger.showSnackBar(const SnackBar(
+              content: Text('無法選擇已劃掉的選項！'),
```

Comment:
```suggestion
              content: Text('無法選擇已劃掉的選項，對選項左右滑動即可取消。'),
```

或許可告知使用者如何取消排除選項的效果。
然後非手機布局（layout）的話就要改成其他提示訊息。

Reply:
OK